docs/reference/search/profile.asciidoc

```diff
@@ -0,0 +1,599 @@
+[[search-profile]]
+== Profile API
+
+coming[2.1.0]
+
+experimental[]
+
+The Profile API provides detailed timing information about the execution of individual components
+in a query.  It gives the user insight into how queries are executed at a low level so that
+the user can understand why certain queries are slow, and take steps to improve their slow queries.
+
+The output from the Profile API is *very* verbose, especially for complicated queries executed across
+many shards. Pretty-printing the response is recommended to help understand the output
+
+[NOTE]
+---------------------
+The details provided by the Profile API directly expose Lucene class names and concepts, which means
+that complete interpretation of the results require fairly advanced knowledge of Lucene.  This
+page attempts to give a crash-course in how Lucene executes queries so that you can use the Profile API to successfully
+diagnose and debug queries, but it is only an overview.  For complete understanding, please refer
+to Lucene's documentation and, in places, the code.
+
+With that said, a complete understanding is often not required to fix a slow query.  It is usually
+sufficient to see that a particular component of a query is slow, and not necessarily understand why
+the `advance` phase of that query is the cause, for example.
+---------------------
```

Comment:
Maybe also add a note that query profiling needs to disable some query optimizations and insert timers in order to work, which can significantly slow down queries. As a consequence, timings may be used to compare relative costs of some parts of the query tree, but can't be used to have a precise timing of how long each part of the query took.